crates/re_space_view/src/view_property_ui.rs

```diff
@@ -43,75 +84,183 @@ pub fn view_property_ui<A: Archetype>(
-This has the same effect as not setting the value in the blueprint at all.")
-                        .on_disabled_hover_text("The property is already unset.")
-                        .clicked() {
-                        ctx.save_empty_blueprint_component_by_name(&blueprint_path, *component_name);
-                        ui.close_menu();
-                    }
-                });
-
-                // TODO(andreas): The next logical thing here is now to save it to the default blueprint!
-                // This should be fairly straight forward except that we need to make sure that a default blueprint exists in the first place.
-            });
-        }
+        list_item::ListItem::new(ctx.re_ui)
+            .interactive(false)
+            .show_hierarchical_with_children(
+                ui,
+                ui.make_persistent_id(archetype.name.full_name()),
+                true,
+                list_item::LabelContent::new(archetype.display_name),
+                sub_prop_ui,
+            );
+    }
+}
+
+/// Draw view property ui for a single component of a view property archetype.
+#[allow(clippy::too_many_arguments)]
+fn view_property_component_ui(
+    ctx: &QueryContext<'_>,
+    ui: &mut egui::Ui,
+    component_name: ComponentName,
+    root_item_display_name: &str,
+    field_info: Option<&ArchetypeFieldInfo>,
+    blueprint_path: &re_log_types::EntityPath,
+    component_results: &re_query::LatestAtComponentResults,
+    fallback_provider: &dyn ComponentFallbackProvider,
+) {
+    let singleline_list_item_content = singleline_list_item_content(
+        ctx,
+        root_item_display_name,
+        blueprint_path,
+        component_name,
+        component_results,
+        fallback_provider,
+    );
+
+    let ui_types = ctx
+        .viewer_ctx
+        .component_ui_registry
+        .registered_ui_types(component_name);
+
+    let mut list_item_response = if ui_types.contains(ComponentUiTypes::MultiLineEditor) {
+        let default_open = false;
+        let id = egui::Id::new((blueprint_path.hash(), component_name));
+        list_item::ListItem::new(ctx.viewer_ctx.re_ui)
+            .interactive(false)
+            .show_hierarchical_with_children(
+                ui,
+                id,
+                default_open,
+                singleline_list_item_content,
+                |_, ui| {
+                    ctx.viewer_ctx.component_ui_registry.multiline_edit_ui(
+                        ctx,
+                        ui,
+                        ctx.viewer_ctx.blueprint_db(),
+                        blueprint_path,
+                        component_name,
+                        component_results,
+                        fallback_provider,
+                    );
+                },
+            )
+            .item_response
+    } else {
+        list_item::ListItem::new(ctx.viewer_ctx.re_ui)
+            .interactive(false)
+            .show_flat(ui, singleline_list_item_content)
```

Comment:
Just to be sure (I've not worked out every code paths in the "registered edit ui kinds" vs. "mono/multi-comp archetype" space), are we sure there is no way that this item might have a hierarchical sibling? It would look mis-aligned in that case.